### ISSUE TITLE
Why Keras behave better and faster than Pytorch under the same network configuration?

### ISSUE THREAD

**Issue by daifeng2016** [pytorch/pytorch#31311]
Recently, I have compared unet++ implementation of Keras version and Pytorch version on the same dataset. However, with Keras the loss decrease continuously and the accuracy is higher after 10 epochs, while with Pytorch the loss decrease unevenly and the  accuracy is lower after 10 epochs.  In my test, I used 'kaiming' initialization and Adam optimization (lr is set to 1e-4) for both keras and pytorch, I have tried my test many times (using torch.backends.cudnn.benchmark=True, num_workers>0, pin_memory=True, etc. ), however, Pytorch version is always much slower,  the training time for keras is about 2984s,  while it is about 4000s for pytorch. Has anyone ever met such problems? 

cc @peterjc123 @VitalyFedyunin @ngimel @mruberry

**Comment by zou3519** [comment]
Could you link the unet++ implementation? Furthermore, could please provide more information about what version of pytorch and keras you are using?   Running and pasting the output of https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py would be great.

**Comment by daifeng2016** [comment]
The unet++implementations are all from github, the pytorch link is  https://github.com/ShawnBIT/UNet-family/blob/master/networks/UNet_Nested.py, while the keras link is https://github.com/MrGiovanni/UNetPlusPlus, . In addition, my pytorch version is 1.0.0, while my keras version is 2.2.4 with tensorflow backend (version is 1.12.0).

**Comment by daifeng2016** [comment]
Below is the environment info:
PyTorch version: 1.0.0
Is debug build: No
CUDA used to build PyTorch: 9.0

OS: Microsoft Windows 10 Pro
GCC version: (MinGW.org GCC-8.2.0-5) 8.2.0
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 390.77
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip3] numpy==1.16.4+mkl
[pip3] numpydoc==0.7.0
[pip3] torch==1.0.0
[pip3] torchfile==0.1.0
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.2.1
[pip3] torchviz==0.0.1
[conda] mkl                       2018.0.0             h36b65af_4  
[conda] mkl-service               1.1.2            py36h57e144c_4  
[conda] numpy                     1.16.4+mkl               pypi_0    pypi
[conda] torch                     1.0.0                    pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.2.1                    pypi_0    pypi
[conda] torchviz                  0.0.1                    pypi_0    pypi

**Comment by alinpahontu2912** [comment]
Hey @daifeng2016 is this issue still relevant with the latest releases of pytorch and keras ?

**Comment by alinpahontu2912** [comment]
I am closing this issue due to lack of activity, feel free to open a new one with needed.
